File: pricemap-python/pricemap/domain/entities/listing.py
from datetime import datetime
from enum import Enum
from typing import Optional


class BuildingType(Enum):
    HOUSE = "HOUSE"
    APARTMENT = "APARTMENT"
    STUDIO = "STUDIO"


class Listing:
    def __init__(
        self,
        id: int,
        name: str,
        description: str,
        building_type: BuildingType,
        street_address: str,
        postal_code: str,
        city: str,
        country_iso_2: str,
        price_eur: int,
        surface_area_m2: int,
        room_count: int,
        bedrooms_count: int,
        contact_phone_number: Optional[str],
        created_date: datetime,
        updated_date: datetime,
    ):
        self.id = id
        self.name = name
        self.description = description
        self.building_type = building_type
        self.street_address = street_address
        self.postal_code = postal_code
        self.city = city
        self.country_iso_2 = country_iso_2
        self.price_eur = price_eur
        self.surface_area_m2 = surface_area_m2
        self.room_count = room_count
        self.bedrooms_count = bedrooms_count
        self.contact_phone_number = contact_phone_number
        self.created_date = created_date
        self.updated_date = updated_date

File: pricemap-dotnet/Models/Listing.cs
using Newtonsoft.Json;
using System;

namespace pricemap.Models
{
    public class Listing
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("postal_address")]
        public PostalAddress PostalAddress { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("building_type")]
        public string BuildingType { get; set; }

        [JsonProperty("price")]
        public Price Price { get; set; }

        [JsonProperty("surface_area_m2")]
        public int SurfaceAreaM2 { get; set; }

        [JsonProperty("rooms_count")]
        public int RoomsCount { get; set; }

        [JsonProperty("bedrooms_count")]
        public int BedroomsCount { get; set; }

        [JsonProperty("contact_phone_number")]
        public string ContactPhoneNumber { get; set; }
    }

    [System.CodeDom.Compiler.GeneratedCode("NJsonSchema", "10.0.22.0 (Newtonsoft.Json v11.0.0.0)")]
    public enum RealEstateListingBuilding_type
    {
        [System.Runtime.Serialization.EnumMember(Value = @"STUDIO")]
        STUDIO = 0,

        [System.Runtime.Serialization.EnumMember(Value = @"APARTMENT")]
        APARTMENT = 1,

        [System.Runtime.Serialization.EnumMember(Value = @"HOUSE")]
        HOUSE = 2,

    }
}
